### PR TITLE
[RLlib] Upgrade RLlib regression test scripts to new testing tool.

### PR DESCRIPTION
**Pull request by sven1977**
This PR translates all existing RLlib release tests to run with the new `Releaser` tool via anyscale.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [x] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/rllib_tests/README.rst [hunk 1]
```diff
@@ -0,0 +1,21 @@
+RLlib Tests
+===========
+
+This directory contains various RLlib release tests.
+
+You should run these tests with the `releaser <https://github.com/ray-project/releaser>`_ tool.
+
+Overview
+--------
+There are ??? kinds of tests:
```
Comment (krfricke) [inline release/rllib_tests/README.rst:10]: Fill out?

File: release/rllib_tests/app_config.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  - pip3 install -U {{ env["RAY_WHEELS"] | default("ray") }}
+  # Install Atari ROMs.
+  - wget http://www.atarimania.com/roms/Roms.rar
+  - unrar x Roms.rar
+  - python -m atari_py.import_roms .
+  #- sudo mkdir -p /data || true
+  #- sudo chown ray:1000 /data || true
+  #- rm -rf /data/classification.parquet || true
+  #- curl -so create_test_data.py https://raw.githubusercontent.com/ray-project/ray/releases/1.3.0/release/xgboost_tests/create_test_data.py
+  #- python create_test_data.py /data/classification.parquet --seed 1234 --num-rows 1000000 --num-cols 40 --num-partitions 100 --num-classes 2
```
Comment (krfricke) [inline release/rllib_tests/app_config.yaml:27]: Remove this?

File: release/rllib_tests/rllib_tests.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    mail: "sven@anyscale.com"
+    slack: "@rllib-team"
+
+  cluster:
+    app_config: app_config.yaml
+    compute_template: 8gpus_64cpus.yaml
+
+  run:
+    timeout: 3600
+    script: unit_gpu_tests/run.sh
```
Comment (krfricke) [inline release/rllib_tests/rllib_tests.yaml:25]: Does this work? Or do we need to use `bash unit_gpu_tests/run.sh`? (Is +x set automatically)?

File: release/rllib_tests/stress_tests/run_stress_tests.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    except ConnectionError:
+        ray.init()
+
+    for i in range(max_num_repeats):
+        # We are done.
+        if len(experiments_to_run) == 0:
+            break
+
+        # Run remaining experiments.
+        trials = run_experiments(experiments_to_run, resume=False, verbose=2)
```
Comment (krfricke) [inline release/rllib_tests/stress_tests/run_stress_tests.py:77]: Shall we add a `ray.tune.utils.release_test_util.ProgressCallback` here? Then we could track some progress after kicking off. Not mandatory.

File: release/rllib_tests/learning_tests/run_learning_tests.py [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    except ConnectionError:
+        ray.init()
+
+    for i in range(max_num_repeats):
+        # We are done.
+        if len(experiments_to_run) == 0:
+            break
+
+        # Run remaining experiments.
+        trials = run_experiments(experiments_to_run, resume=False, verbose=2)
```
Comment (krfricke) [inline release/rllib_tests/learning_tests/run_learning_tests.py:85]: Shall we add a `ray.tune.utils.release_test_util.ProgressCallback` here? Then we could track some progress after kicking off. Not mandatory.

**Review by krfricke — APPROVED** [review]
Looks good to me, left a couple of minor comments. Feel free to ignore. Assuming the tests run and pass on the releaser tool, feel free to merge

File: release/rllib_tests/app_config.yaml (hunk 2 above)
Comment (sven1977) [reply release/rllib_tests/app_config.yaml:27]: +1

Comment (sven1977) [reply release/rllib_tests/app_config.yaml:27]: done

File: release/rllib_tests/rllib_tests.yaml (hunk 3 above)
Comment (sven1977) [reply release/rllib_tests/rllib_tests.yaml:25]: done

File: release/rllib_tests/README.rst (hunk 1 above)
Comment (sven1977) [reply release/rllib_tests/README.rst:10]: fixed

File: release/rllib_tests/stress_tests/run_stress_tests.py (hunk 4 above)
Comment (sven1977) [reply release/rllib_tests/stress_tests/run_stress_tests.py:77]: Let's do this for the next release, then. Lot's of changes to the RLlib release tests coming up :)

**Pull request merged**